src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,354 @@
+    {
+        var organization = await _organizationRepository.GetByIdAsync(update.OrganizationId);
+
+        if (organization == null)
+        {
+            throw new NotFoundException();
+        }
+
+        var newSeatCount = organization.SmSeats.HasValue
+            ? organization.SmSeats.Value + update.SeatAdjustment
+            : update.SeatAdjustment;
+
+
+        if (update.MaxAutoscaleSeats.HasValue && newSeatCount > update.MaxAutoscaleSeats.Value)
+        {
+            throw new BadRequestException("Cannot set max seat autoscaling below seat count.");
+        }
+
+        var newServiceAccountCount = organization.SmServiceAccounts.HasValue
+            ? organization.SmServiceAccounts + update.ServiceAccountsAdjustment
+            : update.ServiceAccountsAdjustment;
+
+        if (update.MaxAutoscaleServiceAccounts.HasValue && newServiceAccountCount > update.MaxAutoscaleServiceAccounts.Value)
+        {
+            throw new BadRequestException("Cannot set max service account autoscaling below service account count.");
+        }
+
+        if (update.SeatAdjustment != 0)
+        {
+            await AdjustSeatsAsync(organization, update.SeatAdjustment);
+        }
+
+        if (update.ServiceAccountsAdjustment != 0)
+        {
+            await AdjustServiceAccountsAsync(organization, update.ServiceAccountsAdjustment.GetValueOrDefault());
+        }
+
+        if (update.MaxAutoscaleSeats != organization.MaxAutoscaleSmSeats)
+        {
+            UpdateSeatsAutoscaling(organization, update.MaxAutoscaleSeats);
+        }
+
+        if (update.MaxAutoscaleServiceAccounts != organization.MaxAutoscaleSmServiceAccounts)
+        {
+            UpdateServiceAccountAutoscaling(organization, update.MaxAutoscaleServiceAccounts);
+        }
+
+        await _organizationService.ReplaceAndUpdateCacheAsync(organization);
+
+        if (organization.SmSeats.HasValue && organization.MaxAutoscaleSmSeats.HasValue && organization.SmSeats == organization.MaxAutoscaleSmSeats)
+        {
+            await SendEmailAsync(organization, organization.MaxAutoscaleSmSeats.Value, "Seats");
+        }
+
+        if (organization.SmServiceAccounts.HasValue && organization.MaxAutoscaleSmServiceAccounts.HasValue && organization.SmServiceAccounts == organization.MaxAutoscaleSmServiceAccounts
+            && update.ServiceAccountsAdjustment > 0)
+        {
+            await SendEmailAsync(organization, organization.MaxAutoscaleSmServiceAccounts.Value, "Service Accounts");
+        }
+    }
+
+    private async Task SendEmailAsync(Organization organization, int MaxAutoscaleValue, string adjustingProduct)
+    {
+        try
+        {
+            var ownerEmails = (await _organizationUserRepository.GetManyByMinimumRoleAsync(organization.Id,
+                    OrganizationUserType.Owner))
+                .Where(u => u.AccessSecretsManager)
+                .Select(u => u.Email).Distinct();
```

Comment:
@r-tome we've had problems where orgs can be ownerless if they have a provider, and that makes some email sends throw because there are no owners to send emails to. Is there anything we need to do here to avoid that issue?

@cyprain-okeke owners can manage the subscription even if they don't have access to SM personally, so I think you can delete the `Where` condition.

Reply:
Thanks for the insight